Automattic/wp-calypso
# Plans: Add new copy for tier 1 & 2 storage upgrades

Resolves `1200412004370260-as-1201090224957164`.

#### Changes proposed in this Pull Request

* Create new product card copy (headers, feature items, etc.) for storage upgrade "products."
* Create new hooks to determine how (and which) storage upgrades should be displayed for a given site.
* Create a selector to retrieve all of a given site's purchases that are eligible for upgradable storage.
* Modify the storage upgrade pricing page to use the new cards and copy.
* Refactor `getSlugInTerm` so that it can address more use cases.
* Remove the `getTieredProducts` hook, as it is no longer in use.

##### Known issues

* Feature highlighting is not enabled yet, but will be addressed in #56595.
* Some page styles are incorrect on narrow screens. We'll address this in a separate PR.

#### Testing instructions

**NOTE:** The `jetpack/only-realtime-products` feature flag must be enabled during the entire testing process. Append `?flags=jetpack/only-realtime-products` to your browser URL, or enable the flag in your environment's Calypso configuration file.

In both Calypso Blue and Calypso Green, test the following:

1. Select a site with no purchases that qualify for storage upgrades, then visit `/pricing/storage/{siteSlug}`. You should see a screen that resembles the below screenshot. Verify all copy looks correct, all links redirect to checkout for tiered Backup products, and all links are correct for the billing term you have selected (i.e., Monthly or Yearly).
2. Select a site with a NON-TIERED Backup or Security purchase (e.g., Backup Daily, Security Real-time). When you visit `/pricing/storage/{siteSlug}`, you should see the same page as in step 1, and it should function identically.

<img width="462" alt="Screen Shot 2021-10-01 at 10 57 23" src="https://user-images.githubusercontent.com/670067/135652876-78cb6135-e7fc-47b9-9a83-a15b0d25b947.png">

3. Select a site with a TIER 1 Backup (Monthly) purchase. When you visit `/pricing/storage/{siteSlug}`, you should see a screen that resembles the below screenshot. Verify:
  a. All copy looks correct;
  b. The 20GB card is displayed with "You own this product;"
  c. Toggling the billing term to Monthly changes the item pricing, and changes 20GB call-to-action link to say "Manage Subscription" and sends you to the Manage Purchase page for your purchase;
  d. Toggling the billing term to Yearly changes the item pricing, and changes the 20GB call-to-action button to say "Upgrade to Yearly" and sends you to checkout for Backup 20GB with a yearly billing term;
  e. The 1TB card is displayed with the price for Backup 1TB;
  f. Clicking the 1TB card's call-to-action link sends you to checkout for Backup (1TB), with the billing term you selected.

<img width="452" alt="Screen Shot 2021-10-01 at 11 01 03" src="https://user-images.githubusercontent.com/670067/135652906-8e9536e7-3528-4938-9302-d7842f8d8b6d.png">

4. Select a site with a TIER 1 Security (Monthly) purchase. When you visit `/pricing/storage/{siteSlug}`, you should see a screen that resembles the screen from step 3. Verify:
  a. Toggling the billing term to Monthly changes the item pricing, and changes the 20GB card's call-to-action link to say "Manage Subscription" and sends you to the Manage Purchase page for your Security purchase;
  b. Toggling the billing term to Yearly changes the item pricing, and changes the 20GB card's call-to-action link to say "Upgrade to Yearly" and sends you to checkout for Security 20GB with a yearly billing term;
  c. The 1TB card is displayed with the price for Security (1TB), and its call-to-action link sends you to checkout for Security (1TB), with the billing term you selected.

5. Select a site with a TIER 2 Monthly purchase (either Backup or Security). When you visit `/pricing/storage/{siteSlug}`, you should see a screen that resembles the below screenshot. Verify:
  a. The 20GB card is not displayed;
  b. The 1TB card is displayed with "You own this product";
  c. Toggling the billing term to Monthly changes the 1TB card's call-to-action link to say "Manage Subscription" and sends you to the Manage Purchase page for the product you purchased;
  d. Toggling the billing term to Yearly changes the 1TB card's call-to-action link to say "Upgrade to Yearly" and sends you to checkout for the yearly version of the 1TB product you originally purchased;

<img width="455" alt="image" src="https://user-images.githubusercontent.com/670067/135652827-b612f639-115c-4f53-a41d-d7d9e09a75f6.png">
